### PR TITLE
{bio}[foss/2021b] infercnvpy v0.4.0 w/ Python 3.9.6

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
requires:
* #16708

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3106.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/31edc2b46421e82e6f6e7610f04797db for a full test report.

File: easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        'checksums': ['559d4d36b0bd5d4494f925cab3c00cd969783ebb6408fa025df92663965834b8'],
+    }),
+    ('pyreadr', '0.4.7', {
+        'checksums': ['901110d62b4bedaef288f4db81425fb696edc721fe2c34c1083f5fb11050a73c'],
+    }),
+    ('tqdm', '4.64.1', {
+        'checksums': ['5f4f682a004951c1b450bc753c710e9280c5746ce6ffedee253ddbcbf54cf1e4'],
+    }),
+    (name, version, {
+        'preinstallopts': """sed -i 's/^build-backend = .*/build-backend = "setuptools.build_meta"/g' pyproject.toml && sed -i 's/^requires = .*/requires = ["setuptools"]/g' pyproject.toml && sed -i 's/dynamic = \["version"\]/version = "%(version)s"/g' pyproject.toml && """,
```
Comment (jfgrimm) [inline easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb:40]:
```suggestion
        'preinstallopts': local_preinstallopts,
```

File: easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    ('scanpy', '1.9.1'),
+]
+
+use_pip = True
+
+# avoid hatcling requirement to install infercnvpy
+# (since installing it introduces conflicting version requirements with poetry included with Python)
+local_preinstallopts = """sed -i 's/^build-backend = .*/build-backend = "setuptools.build_meta"/g' pyproject.toml && """
+local_preinstallopts += """sed -i 's/^requires = .*/requires = ["setuptools"]/g' pyproject.toml && """
+local_preinstallopts += """sed -i 's/dynamic = \["version"\]/version = "%(version)s"/g' pyproject.toml && """
```
Comment (jfgrimm) [inline easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb:24]: could just use one `sed` call with `-e` for each pattern
```suggestion
local_preinstallopts = """sed -i -e 's/^build-backend = .*/build-backend = "setuptools.build_meta"/g' """
local_preinstallopts += """ -e 's/^requires = .*/requires = ["setuptools"]/g' """
local_preinstallopts += """ -e 's/dynamic = \["version"\]/version = "%(version)s"/g' pyproject.toml && """
```

**Review by jfgrimm — DISMISSED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3125.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/96e7be6208ed1d35a566a26e055a840f for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+dependencies = [
+    ('Python', '3.9.6'),
+    ('SciPy-bundle', '2021.10'),
+    ('leidenalg', '0.8.8'),
+    ('scanpy', '1.8.2'),
+]
+
+use_pip = True
+
+# avoid hatcling requirement to install infercnvpy
```
Comment (bedroge) [inline easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb:20]: small typo:
```suggestion
# avoid hatchling requirement to install infercnvpy
```

**Review by bedroge — APPROVED** [review]
lgtm

File: easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb (hunk 2 above)
Comment (boegel) [reply easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb]: fixed in 6c51b07

File: easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/i/infercnvpy/infercnvpy-0.4.0-foss-2021b.eb]: fixed in 6c51b07

**Comment by bedroge** [comment]
Test report by @bedroge
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bob-Latitude-5300 - Linux Ubuntu 22.04, x86_64, Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz, Python 3.10.6
See https://gist.github.com/32cdb0e2e74d85c91e3f706a6ccc063d for a full test report.

**Comment by bedroge** [comment]
Going in, thanks @boegel!

**Pull request merged**